modules/openthread/platform/diag.c

```diff
@@ -68,3 +75,159 @@ void otPlatDiagAlarmCallback(otInstance *aInstance)
 {
 	ARG_UNUSED(aInstance);
 }
+
+/*
+ * To enable gpio diag commands, in Devicetree create `gpio_ports` node in root path
+ * with `compatible = "gpio-ports"` property and `gpio-ports-phandles` property,
+ * which should contain phandles of GPIO port nodes. e.g:
+ *
+ * gpio-ports {
+ *		compatible = "gpio-ports";
```

Comment:
If this is something specific to OT, this compatible name is too generic. And this is actually not a hardware description, but a software choice for openthread. @mbolivar-nordic advice here? Can `chosen` node props take phandles?

Reply:
The point of a generic name was that this binding can be used in other places. For example for drivers/gpio/gpio_shell.c to replace/extend  current implementation which uses device_get_binding() (which requires knowledge of node's full name) to get device struct.

Reply:
@gmarull I've done some additional research about moving phandles to `chosen` node. I think that it is not possible. According to documentation: https://docs.zephyrproject.org/apidoc/latest/group__devicetree-generic-chosen.html `DT_CHOSEN` macro returns a "node identifier for the chosen node property" (singular). I tried to put the phandles into a `chosen` property in overlay file anyways and got a warning: `Properties in chosen must be references to nodes`. Trying to use `DT_FOREACH_PROP_ELEM_SEP` macro on `chosen` node results in an error at build time.

Reply:
> @mbolivar-nordic advice here? Can chosen node props take phandles?

As discovered already, no.

Otherwise it's a bindings question and I'm not involved in reviewing bindings at the moment; ask @galak